File: springapp/src/main/java/com/examly/springapp/service/OrderService.java
package com.examly.springapp.service;

import java.util.List;
import com.examly.springapp.model.Order;

public interface OrderService {
    Order addOrder(Long userId, Long cartId, Long addressId);
    void deleteOrder(Long orderId);
    Order getOrderById(Long orderId);
    Order updateShippingAddress(Long orderId,Long addressId);
    List<Order> getOrdersByUser(Long userId);
    List<Order> getOrdersByStatus(Order.OrderStatus status);
    List<Order> getAllOrders();
    Order updateOrderStatus (Long orderId, String status);
}

File: frontend/src/app/services/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Order } from '../models/order.model';
import { environment } from 'src/environments/environment';


@Injectable({
  providedIn: 'root'
})

export class OrderService {

  private apiUrl = `${environment.apiUrl}/orders`;

  constructor(private http: HttpClient) {}

  placeOrder(order: {userId:Number, cartId:Number, addressId:Number}): Observable<Order> {
    console.log(order);
    return this.http.post<Order>(`${this.apiUrl}/user/${order.userId}/cart/${order.cartId}`, {addressId:order.addressId});
  }
  //
  getUserOrders(userId: number): Observable<Order[]> {
    return this.http.get<Order[]>(`${this.apiUrl}/user/${userId}`);
  }

  getAllOrders(): Observable<Order[]> {
    return this.http.get<Order[]>(`${this.apiUrl}`);
  }

  updateOrderStatus(orderId: number, status: string): Observable<any> {
    console.log(status)
    return this.http.put(`${this.apiUrl}/${orderId}/status`, { status:status });
  }

}
